packages/extension-driver-duckdb/src/lib/duckdbExtensionLoader.ts

```diff
@@ -37,29 +37,38 @@ export class DuckDBExtensionLoader {
       return;
     }
     try {
-      conn.run(`LOAD ${extensionName}`);
+      await new Promise<void>((resolve, reject) => {
+        conn.run(`LOAD ${extensionName}`, (err: any) => {
+          if (err) reject(err);
+          this.logger.debug('Extension loaded');
+          resolve();
+        });
+      });
     } catch (error) {
       this.logger.debug(`Error when loading extension:${extensionName}`);
       throw error;
     }
 
     Object.entries(extensionConfigurations).forEach(
-      ([dbParameterName, configurationKey]) => {
+      async ([dbParameterName, configurationKey]) => {
```

Comment:
Unable to use the `async` in the `forEach` method, need to change to `Promise.all` and `map`